### ISSUE TITLE
build broken on JVM

### ISSUE THREAD

**Issue by coke** [rakudo/rakudo#5337]
Can see it in the azure pipelines, also fails here locally:

```
+++ Compiling	rakudo.jar
+++ Generating	gen/jvm/ast.nqp
+++ Generating	gen/jvm/BOOTSTRAP/v6c.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6c.jar
+++ Compiling	blib/CORE.c.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.000
Impossible coercion from BOOTInt into Int: value is of unacceptable type BOOTInt
  in throw_or_die (gen/jvm/Metamodel.nqp:51)
  in !invalid (gen/jvm/Metamodel.nqp:5339)
  in !invalid_type (gen/jvm/Metamodel.nqp:5354)
  in !coerce_TargetType (gen/jvm/Metamodel.nqp:5322)
  in coerce (gen/jvm/Metamodel.nqp:5308)
  in indent (gen/jvm/CORE.c.setting:18050)
  in <anon> (gen/jvm/rakudo.nqp:151)
  in indent
  in indent
  in descend
  in descend
  in descend
  in <anon>
  in trim_heredoc
  in <anon>
  in <anon>
  in heredoc
  in _ws
  in ws
  in statementlist
  in blockoid
  in method_def
  in routine_declarator:sym<method>
  in !protoregex
  in routine_declarator
  in term:sym<routine_declarator>
  in !protoregex
  in term
  in termish
  in EXPR
  in EXPR
  in statement
  in statementlist
  in blockoid
  in package_def
  in package_declarator:sym<class>
  in !protoregex
  in package_declarator
  in scoped
  in scope_declarator:sym<my>
  in !protoregex
  in scope_declarator
  in term:sym<scope_declarator>
  in !protoregex
  in term
  in termish
  in EXPR
  in EXPR
  in statement
  in statementlist
  in LANG
  in FOREIGN_LANG
  in comp_unit
  in TOP
  in parse
  in parse
  in execute_stage
  in run
  in <anon>
  in compile
  in eval
  in evalfiles
  in command_eval
  in command_eval
  in command_line
  in MAIN
  in <mainline>
  in <anon>
make: *** [blib/CORE.c.setting.jar] Error 1
```

**Comment by coke** [comment]
Looks like this was fixed in c12f478a1

**Comment by coke** [comment]
Failure mode has changed. it gets further, but dies with
```
+++ Generating	gen/jvm/BOOTSTRAP/v6d.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6d.jar
+++ Compiling	blib/CORE.d.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.000
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (metaspaceArena.cpp:88), pid=8222, tid=8451
#  guarantee(requested_word_size <= chunklevel::MAX_CHUNK_WORD_SIZE) failed: Requested size too large (616370) - max allowed size per allocation is 524288.
#
# JRE version: Java(TM) SE Runtime Environment (20.0.2+9) (build 20.0.2+9-78)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (20.0.2+9-78, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, parallel gc, bsd-aarch64)
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/coke/sandbox/rakudo-jvm/hs_err_pid8222.log
[3.443s][warning][os] Loading hsdis library failed
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#
make: *** [blib/CORE.d.setting.jar] Abort trap: 6
```

**Comment by coke** [comment]
the referenced log file is 5K+ lines, let me know if any of it will be helpful

**Comment by MasterDuke17** [comment]
I believe the openjdk issue (https://bugs.openjdk.org/browse/JDK-8294677) mentioned in https://github.com/rakudo/rakudo/issues/4952 is the cause.Sent from my iPhoneOn Sep 6, 2023, at 11:01 PM, Will Coleda ***@***.***> wrote:﻿
the referenced log file is 5K+ lines, let me know if any of it will be helpful

**Comment by usev6** [comment]
@MasterDuke17 Good catch with the reference to https://github.com/rakudo/rakudo/issues/4952 and the OpenJDK issue.

The build problem originally reported in this issue (#5337) (```Impossible coercion from BOOTInt into Int: value is of unacceptable type BOOTInt```) as been fixed with https://github.com/rakudo/rakudo/pull/5365.

The CI jobs (which use OpenJDK 11) are green again. So I'd suggest to close this issue (as resolved) and keep the other one (#4952) open for the internal crash observed with OpenJDK 17.

@coke Would you be fine with that?

**Comment by coke** [comment]
Sure.
